mmpose/apis/inference_video.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+import os
+
+import numpy as np
+import torch
+from mmcv.parallel import collate, scatter
+from PIL import Image
+
+from mmpose.apis.inference import _box2cs, _xywh2xyxy, _xyxy2xywh
+from mmpose.datasets.dataset_info import DatasetInfo
+from mmpose.datasets.pipelines import Compose
+from mmpose.utils.hooks import OutputHook
+
+os.environ['KMP_DUPLICATE_LIB_OK'] = 'TRUE'
+
+
+def inference_top_down_video_pose_model(model,
+                                        frames_or_paths,
+                                        person_results=None,
+                                        bbox_thr=None,
+                                        format='xywh',
+                                        dataset_info=None,
+                                        return_heatmap=False,
+                                        outputs=None):
+    # only two kinds of bbox format is supported
+    assert format in ['xyxy', 'xywh']
+    assert isinstance(frames_or_paths, (list, tuple))
+
+    # get dataset info
+    dataset_info = DatasetInfo(model.cfg.dataset_info)
+
+    pose_results = []
+    returned_outputs = []
+
+    if person_results is None:
+        # create dummy person results
+        if isinstance(frames_or_paths[0], str):
+            width, height = Image.open(frames_or_paths[0]).size
+        else:
+            height, width = frames_or_paths[0].shape[:2]
+        person_results = [{'bbox': np.array([0, 0, width, height])}]
+
+    # if
+    if len(person_results) == 0:
+        return pose_results, returned_outputs
+
+    # Change for-loop preprocess each bbox to preprocess all bboxes at once.
+    bboxes = np.array([box['bbox'] for box in person_results])
+
+    # Select bboxes by score threshold
+    if bbox_thr is not None:
+        assert bboxes.shape[1] == 5
```

Comment:
Is it consistent with line 116?

Reply:
I think they are consistent. 
If the `bbox_thr` is not None, then the `score` must appear in the bbox to filter the boxes, which make its length to 5. 
If `bbox_thr` is None, then we don't have to filter the boxes, in this case the length of a box can be 4 or 5.

Reply:
I will add more descriptions in the docstring.